### ISSUE TITLE
Policy based configuration bypassed on Debian Jessie

### ISSUE THREAD

**Issue by mtds** [gmr/pgsql-listen-exchange#15]
Hello.

I am having troubles using a specific configuration for pgsql-listen-exchange other
than the one provided by  default with the plugin.

Despite the following configuration defined under `/etc/rabbitmq/rabbitmq.config`:

```
[{pgsql_listen_exchange,
  [
    {host, "localhost"},
    {port, 5432},
    {dbname, "my_db"},
    {user, "mu_user"},
    {password, "my_pwd"}
  ]}
].
```

it looks like the plugin is always using `user: postgres' without any password.
From the logs of my RabbitMQ server it looks like that the configuration is correctly
acquired:

```
=INFO REPORT==== 4-Sep-2015::16:44:37 ===
node           : rabbit@pgsqlsrv
home dir       : /var/lib/rabbitmq
config file(s) : /etc/rabbitmq/rabbitmq.config
[...]

=INFO REPORT==== 4-Sep-2015::16:46:53 ===
Server startup complete; 8 plugins started.
 * amqp_client
 * epgsql
 * mochiweb
 * pgsql_listen_exchange
[...]
```

But whenever I have tried to define a RabbitMQ exchange, like this:

```
rabbitmqadmin declare exchange name=testpgsql type=x-pgsql-listen
```

I will get the following error:

```
*** 506 pgsql_listen_worker failure (invalid_password)
```

And indeed, from the Postgres log file, I can see:

```
2015-09-05 23:19:23 CEST [31316-1] postgres@postgres FATAL:  password authentication failed for user "postgres"
2015-09-05 23:19:23 CEST [31316-2] postgres@postgres DETAIL:  User "postgres" has no password assigned.
        Connection matched pg_hba.conf line 92: "host    all             all             127.0.0.1/32  
```

I have also tried to change the configuration file shipped with the plugin (`include/pgsql_listen.hrl`)
but the result is always the same.

I am using Erlang/OTP 17.3, RabbitMQ 3.3.5 and Postgresql 9.4 (all came packaged with Debian 8).

Thanks in advance!

**Comment by gmr** [comment]
Are you able to connect to PostgreSQL with psql using the password specified using the following command?

```bash
  psql -h localhost -p 5432 -d my_db -U mu_user -W 
```

**Comment by mtds** [comment]
Yes, I thought about that but the connection with Postgres is perfectly fine with my 
username/password using ```psql``` on the cmd line:

```
:~$ psql -h localhost -p 5432 -d my_db -U my_user -W
Password for user my_user: 

psql (9.4.3)
SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
Type "help" for help.

my_db =>
```

Here is an extract from ```/etc/postgresql/9.4/main/pg_hba.conf```:

```
# "local" is for Unix domain socket connections only
local   all             all                                     peer
# IPv4 local connections:
host    all             all             127.0.0.1/32            md5
```

RabbitMQ is running on the same host as Postgresql. Since a direct connection with psql works IMHO the problem should not be on the DB authorization side.

**Comment by gmr** [comment]
Ok, just wanted to double check before I start trying to recreate. Can you try configuring via a policy instead of via the rabbitmq.config and see if you have the same issue? We use policies in production and it might be the key difference there.

Also as a heads up I've been working on the next version and expect it to drop in the next few days which supports reconnection and a few new features.

**Comment by mtds** [comment]
Thanks for the heads up about the new version, I will keep an eye on it.

@Policies: I am afraid I have choosen the wrong terms for this issue: my main problem 
is creating an exchange of type 'x-pgsql-listen', which I believe should be the first step
to use your plugin. Am I wrong?

I do have a very basic knowledge about RabbitMQ but as far as I understood from the
documentation (https://www.rabbitmq.com/parameters.html): *Policies can be used to configure 
the federation plugin, mirrored queues, alternate exchanges, dead lettering, per-queue TTLs, 
and maximum queue length*.

But it seems I cannot use them to change a plugin configuration or apply  specific changes to
an exchange. Most likely I have misunderstood what you wrote but can you point me to the
right direction?

**Comment by mtds** [comment]
I have solved the problem. As you rightly said, defining a policy and apply it to an
exchange let me pubblish the notifications from PgSQL to RabbitMQ.

I was too much in a hurry to read properly the RabbitMQ docs and I did not understood
correctly the parts about exchanges and policies.

The problem with the configuration file persist, though. I still do no have any clues about
this but if the plugin works with the RabbitMQ policy it's fine for me.
